cmd/sftp-server-driver.go

```diff
@@ -168,9 +178,10 @@ func (f *sftpDriver) getMinIOClient() (*minio.Client, error) {
 	}
 
 	return minio.New(f.endpoint, &minio.Options{
-		Creds:     credentials.NewStaticV4(ui.Credentials.AccessKey, ui.Credentials.SecretKey, ""),
-		Secure:    globalIsTLS,
-		Transport: globalRemoteFTPClientTransport,
+		Creds:                credentials.NewStaticV4(ui.Credentials.AccessKey, ui.Credentials.SecretKey, ""),
+		Secure:               globalIsTLS,
+		Transport:            globalRemoteFTPClientTransport,
+		CustomRequestHeaders: headers,
```

Comment:
```suggestion
		CustomRequestHeaders: headers,
```

These are things we cannot support use SetAppInfo() and write your own RoundTripper override and add the x-forwarded-for or x-real-ip. 

Let MinIO team do the clean work here.

Reply:
@harshavardhana I tried to ask in https://github.com/minio/minio/issues/19698#issuecomment-2102033627 how MinIO prefer to get this implemented but didn't got any answers.

I'm not really familiar with `RoundTripper` but can look about it.
However, why you closed this PR? Now I cannot update it.

Reply:
I closed it because we will send it, there are code changes that you are making here are not correct.

- passing custom headers by extending minio-go
- not using existing APIs for user agent

Reply:
OK but because of GitHub limitations, I cannot re-open this and neither can you if I force push corrected code to closed PR.

Of course I can always create new PR but most of the projects want to avoid having multiple PRs about same topic.